### PR TITLE
net: change `net_time_t` defined type.

### PR DESCRIPTION
**Pull request by canisLupus1313**
Commit changes `net_time_t` to `uint64_t` and adjust related defines. The reason for this change is that in whole Zephyr time is fedined as unsigned integers.

### CODE REVIEW

File: include/zephyr/net/net_time.h [hunk 1]
```diff
@@ -30,14 +30,7 @@ extern "C" {
```
Comment (jukkar) [inline include/zephyr/net/net_time.h:30]: This is not really an openthread change so commit message could just be `net: Change net_time_t defined type`

File: include/zephyr/net/net_time.h [hunk 2]
```diff
@@ -50,61 +43,14 @@ extern "C" {
...
- * is the ideal intermediate building block for timed network protocols.
  */
-typedef int64_t net_time_t;
+typedef uint64_t net_time_t;
 
 /** The largest positive time value that can be represented by net_time_t */
-#define NET_TIME_MAX INT64_MAX
+#define NET_TIME_MAX UINT64_MAX
 
 /** The smallest negative time value that can be represented by net_time_t */
```
Comment (jukkar) [inline include/zephyr/net/net_time.h:52]: The comment is not valid after this change

File: include/zephyr/net/net_time.h [hunk 3]
```diff
@@ -50,61 +43,14 @@ extern "C" {
...
- * maintains the specified contract including all protocol specific tolerances
- * and therefore clients can rely on common semantics of this type. This makes
- * times coming from different hardware peripherals and even from different
- * network nodes comparable within well-defined limits and therefore net_time_t
- * is the ideal intermediate building block for timed network protocols.
  */
-typedef int64_t net_time_t;
+typedef uint64_t net_time_t;
 
 /** The largest positive time value that can be represented by net_time_t */
```
Comment (jukkar) [inline include/zephyr/net/net_time.h:49]: and here too

File: include/zephyr/net/net_time.h [hunk 4]
```diff
@@ -50,61 +43,14 @@ extern "C" {
  * signal. Be aware that while syntonized clocks share the same frequency and
  * phase, they do not usually share the same epoch (zero-point).
  *
- * This also explains why network time, if represented as a cycle value of some
```
Comment (jukkar) [inline include/zephyr/net/net_time.h:53]: You are removing lot of documentation, is there a reason for that?

**Review by jukkar — COMMENTED** [review]
The `net_time_t` was introduced 6 months ago and this PR already changes it in a major way. The `int64_t` used earlier was used to present past time, so don't we need that any more?

**Review by rlubos — COMMENTED** [review]
I miss the information what problem are we trying to solve here, what was the issue with `int64_t` representation?

> The reason for this change is that in whole Zephyr time is fedined as unsigned integers.

Is that really a valid assumption? `k_uptime_get()`/`k_uptime_ticks()` for example both operate on signed `int64_t`.

File: include/zephyr/net/net_time.h (hunk 2 above)
Comment (canisLupus1313) [reply include/zephyr/net/net_time.h]: fixed.

Comment (canisLupus1313) [reply include/zephyr/net/net_time.h]: fixed.

File: include/zephyr/net/net_time.h (hunk 3 above)
Comment (canisLupus1313) [reply include/zephyr/net/net_time.h]: fixed.

File: include/zephyr/net/net_time.h (hunk 1 above)
Comment (canisLupus1313) [reply include/zephyr/net/net_time.h:30]: fixed.

**Comment by canisLupus1313** [comment]
@rlubos 

> Is that really a valid assumption? `k_uptime_get()`/`k_uptime_ticks()` for example both operate on signed `int64_t`.

That's correct but eg both PTP and IEEE drivers use uint as a representation of time. So defining the net time in `int64_t` is kind of incorrect.

> I miss the information what problem are we trying to solve here, what was the issue with int64_t representation?

We actually can compare time between the Protocols mentioned above. Previously it wasn't the case since compared signed to unsigned in C. Plus we don't need to check any validity of type normal to unsigned when calling functions.

**Review by maciejbaczmanski — DISMISSED** [review]

**Review by pdgendt — DISMISSED** [review]
These changes are done on a public header, and should adhere to the breaking API process IMO:
https://docs.zephyrproject.org/latest/develop/api/api_lifecycle.html#introducing-breaking-api-changes

**Comment by canisLupus1313** [comment]
@pdgendt  this API is still experimental https://docs.zephyrproject.org/latest/develop/api/api_lifecycle.html#experimental So the above shouldn't apply.

**Review by maciejbaczmanski — APPROVED** [review]

**Comment by rlubos** [comment]
Personally, I'm not quite convinced with this change. `int64_t` has capacity to store up to almost 300 years of timespan represented in ns (which seems more than enough) and by switching to `uint64_t` we're stripping the type from possibility to represent relative times in the past (negative values). 
IMO if some user of `net_time_t` cannot handle negative values, it should be addressed in there, instead of changing the type globally.

**Comment by jukkar** [comment]
> Personally, I'm not quite convinced with this change. `int64_t` has capacity to store up to almost 300 years of timespan represented in ns (which seems more than enough) and by switching to `uint64_t` we're stripping the type from possibility to represent relative times in the past (negative values).
> IMO if some user of `net_time_t` cannot handle negative values, it should be addressed in there, instead of changing the type globally.

I agree. There was lot of effort from @fgrandel to introduce this feature recently with some use cases in mind.
I think we need better reasoning than just `...whole Zephyr time is fedined as unsigned integers.`, so why do we need to change this, does something break if we do not change this, and more specifically what breaks if we introduce this change.

**Pull request closed without merging**